src/NuGet.Services.AzureSearch/Auxiliary2AzureSearch/DownloadSetComparer.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using NuGet.Services.AzureSearch.AuxiliaryFiles;
+
+namespace NuGet.Services.AzureSearch.Auxiliary2AzureSearch
+{
+    public class DownloadSetComparer : IDownloadSetComparer
+    {
+        private readonly IAzureSearchTelemetryService _telemetryService;
+        private readonly ILogger<DownloadSetComparer> _logger;
+
+        public DownloadSetComparer(
+            IAzureSearchTelemetryService telemetryService,
+            ILogger<DownloadSetComparer> logger)
+        {
+            _telemetryService = telemetryService ?? throw new ArgumentNullException(nameof(telemetryService));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public SortedDictionary<string, long> Compare(
+            DownloadData oldData,
+            DownloadData newData)
+        {
+            var stopwatch = Stopwatch.StartNew();
+
+            // We use a very simplistic algorithm here. Find the union of both ID sets and compare each download count.
+            var uniqueIds = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            foreach (var id in oldData.Keys.Concat(newData.Keys))
```

Comment:
What're your thoughts on instead passing this LINQ `Concat` enumerable to the constructor of the `HashSet`?

Reply:
Same comment for line 83

Reply:
Yeah, great idea! I forgot about that overload.